app/src/main/java/org/mozilla/fenix/components/toolbar/interactor/BrowserToolbarInteractor.kt

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar.interactor
+
+import mozilla.components.ui.tabcounter.TabCounterMenu
+import org.mozilla.fenix.components.toolbar.BrowserToolbarController
+import org.mozilla.fenix.components.toolbar.BrowserToolbarMenuController
+import org.mozilla.fenix.components.toolbar.ToolbarMenu
+
+/**
+ * Interface for the browser toolbar interactor.
+ */
+interface BrowserToolbarInteractor {
+    fun onBrowserToolbarPaste(text: String)
+    fun onBrowserToolbarPasteAndGo(text: String)
+    fun onBrowserToolbarClicked()
+    fun onBrowserToolbarMenuItemTapped(item: ToolbarMenu.Item)
+    fun onTabCounterClicked()
+    fun onTabCounterMenuItemTapped(item: TabCounterMenu.Item)
+    fun onScrolled(offset: Int)
+    fun onReaderModePressed(enabled: Boolean)
+
+    /**
+     * Navigates to the Home screen. Called when a user taps on the Home button.
+     */
+    fun onHomeButtonClicked()
+}
+
+/**
+ * The default implementation of [BrowserToolbarInteractor].
+ *
+ * @property browserToolbarController An instance of [BrowserToolbarController] which will be
+ * delegated for all user interactions on the browser toolbar.
+ * @property menuController An instance of [BrowserToolbarMenuController] which will be delegated
+ * for all user interactions on the the browser toolbar menu.
```

Comment:
I think we could ignore the `An instance of [XX]` as we not that from the data type.

Reply:
Modified the @property description.